File: benchmarks/reference/array_ops.js
function main() {
    const start = process.hrtime();
    let arr = [];
    for (let i = 0; i < 50000; i++) {
        arr.push(i);
    }

    let sum = 0;
    for (let i = 0; i < arr.length; i++) {
        sum += arr[i];
    }

    // Pop half
    for (let i = 0; i < 25000; i++) {
        arr.pop();
    }

    console.log("Array Sum: " + sum);
    console.log("Array Length: " + arr.length);
    const diff = process.hrtime(start);
    const elapsed = diff[0] + diff[1] / 1e9;
    console.log(`Time: ${elapsed.toFixed(6)}`);
}

main();

File: benchmarks/reference/array_ops.py
import time

def main():
    start = time.time()
    arr = []
    for i in range(50000):
        arr.append(i)
    
    total_sum = 0
    for i in range(len(arr)):
        total_sum += arr[i]
    
    # Pop half
    for i in range(25000):
        arr.pop()
    
    print(f"Array Sum: {total_sum}")
    print(f"Array Length: {len(arr)}")
    elapsed = time.time() - start
    print(f"Time: {elapsed:.6f}")

if __name__ == "__main__":
    main()
